matrix-sdk/src/main/java/org/matrix/androidsdk/data/cryptostore/MXFileCryptoStore.java

```diff
@@ -1110,8 +1119,19 @@ private Object loadObject(File file, String description) {
             try {
                 // the files are now zipped to reduce saving time
                 FileInputStream fis = new FileInputStream(file);
-                GZIPInputStream gz = new GZIPInputStream(fis);
+                InputStream cis = CompatUtil.createCipherInputStream(fis, mContext);
+                GZIPInputStream gz;
+
+                if (cis != null) {
+                    gz = new GZIPInputStream(cis);
+                } else {
+                    //fallback to unencrypted stream for backward compatibility
+                    Log.i(LOG_TAG, "## loadObject() : failed to read encrypted, fallback to unencrypted read");
+                    gz = new GZIPInputStream(fis);
```

Comment:
I think that `fis` has been read a bit by `CompatUtil.createCipherInputStream()`, so we may close `fis` and open it again.

Reply:
I can use  `mark`/`reset` which is less agressive; would it be ok ?

Reply:
yes

Reply:
Done. Mark / reset is done in `createCipherInputStream`.